src/rrt/Tree.hpp

```diff
@@ -218,13 +236,23 @@ class Tree {
     }
 
     /**
-     * Removes all Nodes from the tree so it can be run() again.
+     * Removes nodes from _nodes and _nodemap so it can be run() again.
      */
     void reset(bool eraseRoot = false) {
         if (eraseRoot) {
             _nodes.clear();
+            _nodemap.clear();
         } else {
-            if (!_nodes.empty()) {
+            if (_nodes.size() > 1) {
+                auto it = _nodemap.begin();
```

Comment:
oohhhh, in that case, can you rename it to `iter` instead :P. I think that would reduce a lot of confusion 😄